### PR TITLE
VM image content verify

### PR DESCRIPTION
**Pull request by mjudeikis**
```release-note
NONE
```

Script for validating vmimage content RPMS.
FLow:
1. Run validation script on the newest published VM image (based on tags from `images` RG). Follow-up: we need to start marking "live" images and test those. 
2. Get updatable package list and summary
3. Raise an issue
4. Run all this in periodic 

TODO:
* Upload results to private blob storage or appInsights

/cc @jim-minter @asalkeld @kwoodson  @charlesakalugwu

### CODE REVIEW

File: pkg/cluster/gomock_reflect_610001376/prog.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
```
Comment (charlesakalugwu) [inline pkg/cluster/gomock_reflect_610001376/prog.go]: dont think this file should be in here :)

Comment (mjudeikis) [reply pkg/cluster/gomock_reflect_610001376/prog.go]: yes :) will delete

File: pkg/vmimage/resources.go [hunk 2]
```diff
@@ -138,19 +138,33 @@ func nic(subscriptionID, resourceGroup, location string) *network.Interface {
 	}
 }
 
-func vm(subscriptionID, resourceGroup, location string, sshPublicKey string) *compute.VirtualMachine {
+func vm(subscriptionID, resourceGroup, location string, sshPublicKey string, verify bool, image string) *compute.VirtualMachine {
```
Comment (charlesakalugwu) [inline pkg/vmimage/resources.go]: what about changing the signature to:

```suggestion
func vm(subscriptionID, resourceGroup, location, sshPublicKey, image string, verify bool) *compute.VirtualMachine {
```

File: pkg/vmimage/ssh.go [hunk 3]
```diff
@@ -11,11 +11,76 @@ import (
...
+	if err != nil {
+		return err
+	}
+
+	io.Copy(os.Stdout, stdout)
+
+	return nil
+}
+
+func (builder *Builder) scp(files []string) error {
```
Comment (charlesakalugwu) [inline pkg/vmimage/ssh.go]: can we share some of the logic between `*Builder.ssh()` and `*Builder.scp(files []string)`?

Comment (mjudeikis) [reply pkg/vmimage/ssh.go]: We already doing session creation. The only other thing left is `stdoutPipe` but not sure if it worth start passing it around.

**Comment by mjudeikis** [comment]
Example issue: https://github.com/openshift/openshift-azure/issues/1641

**Comment by kwoodson** [comment]
@mjudeikis 

Currently there aren't any results from the `yum` command:
```
[root@vm ~]# cat /tmp/check 
Loaded plugins: product-id, search-disabled-repos, subscription-manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
[root@vm ~]# cat /tmp/info
Loaded plugins: product-id, search-disabled-repos, subscription-manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
updateinfo summary done
```

File: pkg/vmimage/ssh.go [hunk 4]
```diff
@@ -11,11 +11,76 @@ import (
...
+			return err
+		}
+		defer f.Close()
+
+		err = session.Start(fmt.Sprintf("sudo cat %s", file))
+		if err != nil {
+			return err
+		}
+
+		io.Copy(f, stdoutPipe)
```
Comment (asalkeld) [inline pkg/vmimage/ssh.go]: you don't want to check the error?

File: pkg/vmimage/ssh.go [hunk 5]
```diff
@@ -11,11 +11,76 @@ import (
...
 	if err != nil {
 		return err
 	}
 
+	err = session.Start("sudo tail -F -n +1 /var/lib/waagent/custom-script/download/0/stdout")
+	if err != nil {
+		return err
+	}
+
+	io.Copy(os.Stdout, stdout)
```
Comment (asalkeld) [inline pkg/vmimage/ssh.go]: error?

File: pkg/vmimage/ssh.go (hunk 4 above)
Comment (asalkeld) [reply pkg/vmimage/ssh.go]: or is the session going to give you *all* errors, probably not local file errors..

**Comment by asalkeld** [comment]
@mjudeikis looks pretty good, upgrade failing

**Comment by mjudeikis** [comment]
> @mjudeikis
> 
> Currently there aren't any results from the `yum` command:
> 
> ```
> [root@vm ~]# cat /tmp/check 
> Loaded plugins: product-id, search-disabled-repos, subscription-manager
> This system is not registered with an entitlement server. You can use subscription-manager to register.
> [root@vm ~]# cat /tmp/info
> Loaded plugins: product-id, search-disabled-repos, subscription-manager
> This system is not registered with an entitlement server. You can use subscription-manager to register.
> updateinfo summary done
> ```

Because it takes "newest" image, potentially built yesterday. So nothing to update. we need to point it to images, used in production somehow.

File: pkg/vmimage/ssh.go (hunk 4 above)
Comment (mjudeikis) [reply pkg/vmimage/ssh.go]: I was thinking the same. Not sure if my thinking is right

File: pkg/vmimage/ssh.go (hunk 5 above)
Comment (mjudeikis) [reply pkg/vmimage/ssh.go]: same as above?

**Comment by ehashman** [comment]
/retest

upgrade-v5.1 flake #1438 
e2e-create flake #1532 (or at least is similar)

**Pull request merged**